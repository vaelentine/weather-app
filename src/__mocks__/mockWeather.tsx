const mockWeather =
  {
    coord: {
      lon: -122.6762,
      lat: 45.5234
    },
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'clear sky',
        icon: '01n'
      }
    ],
    base: 'stations',
    main: {
      temp: 291.01,
      feels_like: 290.26,
      temp_min: 286.82,
      temp_max: 293.11,
      pressure: 1013,
      humidity: 54
    },
    visibility: 10000,
    wind: {
      speed: 0.89,
      deg: 313,
      gust: 4.02
    },
    clouds: {
      all: 1
    },
    dt: 1630560308,
    sys: {
      type: 2,
      id: 2008548,
      country: 'US',
      sunrise: 1630503161,
      sunset: 1630550935
    },
    timezone: -25200,
    id: 5746545,
    name: 'Portland',
    cod: 200
  }

export default mockWeather
