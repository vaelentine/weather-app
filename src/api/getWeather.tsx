import axios from 'axios'
import * as dotenv from "dotenv";
dotenv.config({ path: __dirname+'/.env' });
// require('dotenv').config()

async function getWeather(city: string) {
  try {
    const server = 'openweathermap.org/'
    const weatherEndpoint = 'data/2.5/weather'
    const keyParam = `appid=${process.env.REACT_APP_WEATHER_API_KEY}`
    const request = `https://api.${server}${weatherEndpoint}?${keyParam}&q=${city}&units=imperial`
    const response = await axios.get(request, {})
    return response
  } catch (error:any) {
    return error.response
  }
}

export default getWeather
