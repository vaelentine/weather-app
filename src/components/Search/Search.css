.Search {
  text-align: center;
  max-width: 680px;
  padding: 100px;
  margin: 100px;
  font-size: large;
}

button {
  cursor: pointer;
  border-radius: 5px;
  padding: 5px 20px;
  font-size: 16px;
  font-weight: bold;
  text-decoration: none;
  margin: 10px;
  color: #fff;
  position: relative;
  display: inline-block;
  background-color: #01223b;
  box-shadow: 0px 5px 0px 0px #1b1e20;
  }
  
  button:active {
    transform: translate(0px, 2px);
    -webkit-transform: translate(0px, 2px);
    box-shadow: 0px 1px 0px 0px;
  }
  
  button:hover {
    background-color: #6FC6FF;
  }


  input {
    font-size: 32px;
    border: none;
    background: none;
    outline: none;
    color: inherit;
    border-bottom: 1px solid;
  }

  .errorMessage {
    color: red;
  }