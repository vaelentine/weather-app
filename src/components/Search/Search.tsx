import React, { useState, useEffect } from 'react'
import './Search.css'
import getWeather from 'api/getWeather'

const Search = ({ response }:any) => {
  const [city, setCity] = useState('')
  return (
    <div className="Search">
      <input
      placeholder="Enter city"
      value={city}
      onChange={(event) => setCity(event.target.value)}
      type="text"
      >
      </input>
      <button
        className="SearchButton"
        onClick={ () => response(getWeather(city))}
      >
        Find out!
      </button>
    </div>
  )
}

export default Search
