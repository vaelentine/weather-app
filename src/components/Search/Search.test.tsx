/* eslint-disable no-undef */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Search from './Search'
import axios from 'axios'
import getWeather from 'api/getWeather'
import mockWeather from '__mocks__/mockWeather'

jest.mock('axios')

describe('Search Component', () => {
  it('renders correctly', () => {
    const component = render(<Search />)
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('accepts an input value', () => {
    const component = render(<Search />)
    // text input component by placeholder value
    const searchInput = component.getByPlaceholderText('Enter city')
    fireEvent.input(searchInput, { target: { value: 'Portland' } })
    expect(searchInput.value).toBe('Portland')
  })

  it('makes an API call when the button is clicked', async () => {
    const component = render(<Search />)
    const searchInput = component.getByPlaceholderText('Enter city')
    fireEvent.input(searchInput, { target: { value: 'Portland' } })
    axios.get.mockImplementation(() => Promise.resolve(mockWeather))
    const searchButton = component.getByText('Find out!')
    fireEvent.click(searchButton)
    // get mock api call
    expect(axios.get).toHaveBeenCalledTimes(1)
  })

  it.skip('displays error correctly', () => {
    // mock a return value for an error state
    mockGetWeather.mockReturnValueOnce({ message: 'API Error' })
    const component = render(<Search />)
    const searchInput = component.getByPlaceholderText('Enter city')
    fireEvent.input(searchInput, { target: { value: 'NotARealCityName' } })
    const filledInput = component.getByText('NotARealCityName')
    expect(filledInput).toBeInTheDocument()

    const searchButton = component.getByText('Find out!')
    fireEvent.click(searchButton)
    // get mock api call
    expect(mockGetWeather).toBeCalledTimes(1)
    expect(component.getByText('API Error')).toBeInTheDocument()
  })
})
 