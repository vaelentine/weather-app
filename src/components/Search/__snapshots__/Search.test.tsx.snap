// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <input
      placeholder="Enter city"
      type="text"
      value=""
    />
    <button
      class="SearchButton"
    >
      Find out!
    </button>
  </div>
</DocumentFragment>
`;
